### PR TITLE
[CI] Add DeepSeek-R1 (with MLA, no accuracy, 12-decoder layers only) to CI test

### PR DESCRIPTION
**Pull request by kwang3939**
# Description

Add an extra test based on the existing one: https://github.com/vllm-project/tpu-inference/blob/18ee19d8b765dbc21bf235e8bfce51b5aae530cd/.buildkite/pipeline_jax.yml#L181 with MLA enabled.

# Tests

Tested locally with `.buildkite/scripts/run_in_docker.sh   bash -c 'export TPU_BACKEND_TYPE=jax; bash /workspace/tpu_inference/tests/e2e/benchmarking/mlperf.sh -m jrplatin/DeepSeek-R1-1D-Subchannel-256 --use-dummy-weights'`
Set env variables: 
```
export IS_FOR_V7X="true"
export NEW_MODEL_DESIGN="1"
export SKIP_ACCURACY_TESTS="True"
export VLLM_MLA_DISABLE="0"
```

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tests/e2e/benchmarking/mlperf.sh [hunk 1]
```diff
@@ -281,8 +281,18 @@ for model_name in $model_list; do
         max_batched_tokens=1024
         if [ "$model_name" == "meta-llama/Llama-4-Scout-17B-16E-Instruct" ]; then
             current_serve_args+=(--hf-overrides '{"architectures": ["Llama4ForCausalLM"]}')
-        elif [ "$model_name" == "jrplatin/DeepSeek-R1-1D-Subchannel-256" ]; then
-            current_serve_args+=(--hf_overrides '{"num_hidden_layers": 12}')
+        fi
+    fi
+
+    if [ "$model_name" == "jrplatin/DeepSeek-R1-1D-Subchannel-256" ]; then
```
Comment (jrplatin) [inline tests/e2e/benchmarking/mlperf.sh:287]: Don't we want this to be guarded by the `USE_V6E8_QUEUE` flag?

File: .buildkite/pipeline_jax.yml [hunk 2]
```diff
@@ -292,6 +292,21 @@ steps:
            exit 0
          fi
 
+   - label: "${LABEL_PREFIX}E2E MLperf tests for DeepSeek-R1 (with MLA, no accuracy, 12-decoder layers only)"
+     key: ${KEY_PREFIX}test_20
+     soft_fail: true
+     env:
+       NEW_MODEL_DESIGN: "1"
+       SKIP_ACCURACY_TESTS: "True"
```
Comment (jrplatin) [inline .buildkite/pipeline_jax.yml:291]: Do we not want `       USE_V6E8_QUEUE: "True"`?

File: tests/e2e/benchmarking/mlperf.sh [hunk 3]
```diff
@@ -281,8 +281,18 @@ for model_name in $model_list; do
...
+        max_batched_tokens=1024
+        current_serve_args+=(--tensor-parallel-size 8)
```
Comment (jrplatin) [inline tests/e2e/benchmarking/mlperf.sh:289]: Is this going to run on Ironwood as well? Is setting this to 8 going to have issues if running on a v7x-2?

**Comment by jrplatin** [comment]
Thanks for this Kewei! Can you please add a link to the passing BuildKite CI run please? Thanks!

File: .buildkite/pipeline_jax.yml (hunk 2 above)
Comment (kwang3939) [reply .buildkite/pipeline_jax.yml:291]: I removed that since this test is for v7 only. So I also changed the logic to move the model out of v6e-8 specific.

File: tests/e2e/benchmarking/mlperf.sh (hunk 3 above)
Comment (kwang3939) [reply tests/e2e/benchmarking/mlperf.sh:289]: I think since the queue is set to `TPU_QUEUE_MULTI`, only v7x-8 will be used based on this line: https://github.com/vllm-project/tpu-inference/blob/18ee19d8b765dbc21bf235e8bfce51b5aae530cd/.buildkite/scripts/bootstrap.sh#L95C5-L95C45
